packages/py-api/py_api/controllers/hackathon_participants_controller.py

```diff
@@ -0,0 +1,81 @@
+import json
+from typing import Any, Dict
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import participants_col
+from py_api.models import NewParticipant, UpdateParticipant
+
+
+class PartcipantsController:
+
+    def filter_none_values(document: UpdateParticipant) -> Dict[str, Any]:
+        # Creates a dictionary for participant_form
+        participant_form_dump = document.model_dump()
+        fields_to_be_updated = {}
+
+        # It pushes the fields whose value is not null to the empty dictionary
+        for key, value in participant_form_dump.items():
+            if value:
+                fields_to_be_updated[key] = value
+
+        return fields_to_be_updated
+
+    def get_all_participants() -> JSONResponse:
+        participants = list(participants_col.find())
+
+        if not participants:
+            return JSONResponse(content={"message": "No participants were found!"}, status_code=404)
+
+        # the dumps funtion from bson.json_util returns a string
+        return JSONResponse(content={"participants": json.loads(dumps(participants))}, status_code=200)
+
+    def get_specified_participant(object_id: str) -> JSONResponse:
+        try:
+            specified_participant = participants_col.find_one(
+                filter={"_id": ObjectId(object_id)},
+            )
+        except:
```

Comment:
Your version is still not working. Test your code!
Here is the correct way to fix it. 
Add this as import:
```python
from bson.errors import InvalidId
```
![image](https://github.com/AUBGTheHUB/monolith/assets/38375079/1b4556ec-b0d7-4bcd-a9bb-e8d0f400ef21)
If you look at the docs of `ObjectId` you will see -> `Raises ~bson.errors.InvalidId if oid is not 12 bytes nor
24 hex digits, or TypeError if oid is not an accepted type.`

Reply:
it's a good idea to have the code pasted rather than as an image